# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/COLRFonts.h
# Commit: e6523049b675
# Full Hash: e6523049b6751146641d019b9adc0825473c46aa
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2022-08-17 09:10:29
# Regressor Bug: 1740530
# File Overlap Count: 1
# Description:
#   Bug 1740530 - patch 1 - Move COLR font support functions from gfxUtils into their own class. r=gfx-reviewers,lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D152036
# ==============================================================================

diff -r 90f3275f3489 -r e6523049b675 gfx/thebes/COLRFonts.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/gfx/thebes/COLRFonts.h	Tue Aug 16 12:48:31 2022 +0000
@@ -0,0 +1,32 @@
+/* -*- Mode: C++; tab-width: 20; indent-tabs-mode: nil; c-basic-offset: 2 -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef COLR_FONTS_H
+#define COLR_FONTS_H
+
+#include "nsTArray.h"
+
+struct hb_blob_t;
+
+namespace mozilla {
+namespace gfx {
+struct DeviceColor;
+
+class COLRFonts {
+ public:
+  // for color layer from glyph using COLR and CPAL tables
+  static bool ValidateColorGlyphs(hb_blob_t* aCOLR, hb_blob_t* aCPAL);
+  static bool GetColorGlyphLayers(
+      hb_blob_t* aCOLR, hb_blob_t* aCPAL, uint32_t aGlyphId,
+      const mozilla::gfx::DeviceColor& aDefaultColor,
+      nsTArray<uint16_t>& aGlyphs,
+      nsTArray<mozilla::gfx::DeviceColor>& aColors);
+  static bool HasColorLayersForGlyph(hb_blob_t* aCOLR, uint32_t aGlyphId);
+};
+
+}  // namespace gfx
+}  // namespace mozilla
+
+#endif  // COLR_FONTS_H